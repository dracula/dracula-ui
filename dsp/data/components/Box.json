{
  "class": "component",
  "type": "page",
  "id": "Box",
  "name": "Box",
  "last_updated": "2021-03-14T00:08:30.329Z",
  "last_updated_by": "System",
  "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "box",
    "languages": {
      "javascriptreact": "<Box color=\"purple\">Dracula</Box>;\n",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n",
      "react": "<Box color=\"purple\">Dracula</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Box.png",
      "svgData": "./dsp/assets/svgs/Box.svg"
    },
    {
      "title": "Colors",
      "html": "<div color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">Pink Purple</div>\n",
      "react": "<Box color=\"pinkPurple\">Pink Purple</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxColors.png",
      "svgData": "./dsp/assets/svgs/BoxColors.svg"
    },
    {
      "title": "Rounded",
      "html": "<div style=\"display: flex; flex-direction: row\">\n  <div\n    color=\"purple\"\n    rounded=\"none\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-none\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"sm\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-sm\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"base\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"lg\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-lg\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"2xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-2xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"3xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-3xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"full\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-full\"\n  ></div>\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"row\" }}>\n  <Box\n    color=\"purple\"\n    rounded=\"none\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"sm\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"base\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"lg\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"2xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"3xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"full\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxRounded.png",
      "svgData": "./dsp/assets/svgs/BoxRounded.svg"
    },
    {
      "title": "Generic Elements",
      "html": "<div>\n  <button as=\"button\" color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">\n    Button</button\n  ><a as=\"a\" color=\"green\" href=\"#\" class=\"drac-box drac-bg-green\">Link</a>\n</div>\n",
      "react": "<div>\n  <Box as=\"button\" color=\"pinkPurple\">\n    Button\n  </Box>\n  <Box as=\"a\" color=\"green\" href=\"#\">\n    Link\n  </Box>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxGeneric Elements.png",
      "svgData": "./dsp/assets/svgs/BoxGeneric Elements.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "color": {
      "defaultValue": null,
      "description": "The background color.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "rounded": {
      "defaultValue": null,
      "description": "The border radius.",
      "name": "rounded",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
      }
    },
    "spacing": {
      "defaultValue": null,
      "description": "Dracula UI standard spacing properties.",
      "name": "spacing",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "SpacingPropType | undefined" }
    },
    "margin": {
      "defaultValue": null,
      "description": "Dracula UI standard margin properties.",
      "name": "margin",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "MarginPropType | undefined" }
    },
    "as": {
      "defaultValue": null,
      "description": "The HTML element to be used",
      "name": "as",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
    "displayName": "Box",
    "methods": [],
    "props": {
      "color": {
        "defaultValue": null,
        "description": "The background color.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
        }
      },
      "rounded": {
        "defaultValue": null,
        "description": "The border radius.",
        "name": "rounded",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
        }
      },
      "spacing": {
        "defaultValue": null,
        "description": "Dracula UI standard spacing properties.",
        "name": "spacing",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "SpacingPropType | undefined" }
      },
      "margin": {
        "defaultValue": null,
        "description": "Dracula UI standard margin properties.",
        "name": "margin",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "MarginPropType | undefined" }
      },
      "as": {
        "defaultValue": null,
        "description": "The HTML element to be used",
        "name": "as",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
      }
    }
  }
}
