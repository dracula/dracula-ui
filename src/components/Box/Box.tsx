import cx from 'classnames/dedupe'
import React, { AllHTMLAttributes } from 'react'
import { borderColors, colors, glowColors } from '../../base/colors'
import {
  PaddingMixin,
  paddingMixin,
  marginMixin,
  MarginMixin
} from '../../base/spacing'

type Element = HTMLElementTagNameMap

export const roundedBorders = {
  none: 'drac-rounded-none',
  sm: 'drac-rounded-sm',
  base: 'drac-rounded',
  lg: 'drac-rounded-lg',
  xl: 'drac-rounded-xl',
  '2xl': 'drac-rounded-2xl',
  '3xl': 'drac-rounded-3xl',
  full: 'drac-rounded-full'
}

/**
 * Box Props
 */
export type BoxProps<K extends keyof Element = 'div'> = {
  /** The background color. */
  color?: keyof typeof colors

  /** The glow color. */
  glowColor?: keyof typeof glowColors

  /** The border color. */
  borderColor?: keyof typeof borderColors

  /** The border radius. */
  rounded?: keyof typeof roundedBorders

  /** The HTML element to be used */
  as?: K
} & AllHTMLAttributes<K> &
  PaddingMixin &
  MarginMixin

/**
 * Box is the most primitive component of Dracula UI.
 * Using Box allows for consumers of the components library to compose
 * more complex patterns, components, and UIs.
 *
 * Box includes built-in Color and Spacing properties that make building
 * complex components convenient and consistent.
 */
export function Box<T extends keyof Element>(props: BoxProps<T>) {
  const finalProps = {
    ...props,
    className: cx(
      `drac-box`,
      props.className,
      props.color && colors[props.color],
      props.glowColor && glowColors[props.glowColor],
      props.borderColor && borderColors[props.borderColor],
      props.rounded && roundedBorders[props.rounded],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  const as = finalProps.as ?? 'div'
  return React.createElement(as, finalProps, props.children)
}

Box.displayName = 'Box'
