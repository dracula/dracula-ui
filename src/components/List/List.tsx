import cx from 'classnames/dedupe'
import React, { HTMLAttributes } from 'react'
import { ColorMap } from '../../base/colors'
import {
  cleanProps,
  marginMixin,
  MarginMixin,
  paddingMixin,
  PaddingMixin
} from '../../base/spacing'

export const listColors: Partial<ColorMap> = {
  white: 'drac-list-white',
  cyan: 'drac-list-cyan',
  green: 'drac-list-green',
  orange: 'drac-list-orange',
  pink: 'drac-list-pink',
  purple: 'drac-list-purple',
  red: 'drac-list-red',
  yellow: 'drac-list-yellow'
}

export const listVariants = {
  unordered: 'drac-list-unordered',
  ordered: 'drac-list-ordered',
  none: 'drac-list-none'
}

/**
 * List Props
 */
export interface ListProps
  extends HTMLAttributes<HTMLUListElement>,
    PaddingMixin,
    MarginMixin {
  /** The Dracula UI color for the List. */
  color?: keyof typeof listColors
  /**
   * The variation to be used for the List element.
   * `unordered` -> Display list in dashes as an `<ul>`
   * `ordered` -> Deprecated and moved to OrderedList
   */
  variant?: keyof typeof listVariants
}

/**
 * Lists are used to display list items in an unordered way.
 */
export const List: React.FC<ListProps> = (props: ListProps) => {
  const { color, variant, ...htmlProps } = props

  if (variant === 'ordered')
    throw new Error(
      'the `ordered` prop has been deprecated. Please use the OrderedList component.'
    )

  const finalProps = cleanProps({
    ...htmlProps,
    className: cx(
      'drac-list',
      props.className,
      variant && listVariants[variant],
      color && listColors[color],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  })

  return <ul {...finalProps} />
}

List.displayName = 'List'
