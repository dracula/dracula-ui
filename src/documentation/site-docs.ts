import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from './render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from './pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'
import os from 'os'

interface Documentation {
  basic: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder> | SnapshotBuilder
}

jest.setTimeout(10000)

const RECORDING = process.env.RECORDING

export function docs<T>(
  componentClass: React.FunctionComponent<T> | string,
  documentation: Documentation
) {
  const name =
    typeof componentClass === 'string'
      ? componentClass
      : componentClass.displayName ?? componentClass.name

  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeAll(async () => {
      if (!RECORDING) {
        return
      }

      let executablePath = undefined
      if (os.arch() === 'arm64') {
        executablePath = '/Applications/Chromium.app/Contents/MacOS/Chromium'
      }

      browser = await puppeteer.launch({
        headless: true,
        devtools: true,
        executablePath
      })
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      if (!RECORDING) {
        return
      }

      await browser.close()
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basic()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      if (RECORDING) {
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          null,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg
      }

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations: SnapshotBuilder[] = ([] as SnapshotBuilder[]).concat(
      documentation.variations()
    )

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        if (RECORDING) {
          const [screenshot, svg] = await componentScreenshot(
            browser,
            snapshot,
            variation,
            name
          )

          snapshot.screenshot = screenshot
          snapshot.svgData = svg
        }

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
