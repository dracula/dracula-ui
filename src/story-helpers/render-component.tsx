import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { pretty } from './pretty'

Enzyme.configure({ adapter: new Adapter() })

export type ComponentExample = {
  html: string
  docs: string
  title: string
  react?: any
  screenshot?: string
  svgData?: string
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): ComponentExample {
  let component = comp()
  const html = mount(component).html()
  let wrapped = false

  // wrap the component in a div when it's not wrapped already
  // we do this so we can get the actual component example to not be shallow rendered.
  // Shallow rendering a parent "div" gives us a nice JSX representation of the example.
  if (component.type !== 'div') {
    component = <div>{component}</div>
    wrapped = true
  }

  let react = shallow(component).debug({ verbose: true })

  if (wrapped) {
    // replace wrapper div ðŸ˜¬
    react = react
      .replace(/^<div>\n/, '')
      .replace(/\n<\/div>$/, '')
      .trim()
  }

  return {
    title,
    html: pretty(html, 'html'),
    react: pretty(react, 'typescript'),
    docs
  }
}
