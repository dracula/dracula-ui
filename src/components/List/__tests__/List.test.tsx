import React from 'react'
import { docs } from '@/documentation/site-docs'
import { snapshot } from '@/documentation/render-component'
import { Box } from '@/components/Box/Box'
import { List } from '@/components/List/List'

docs(List, {
  basic() {
    return snapshot('Usage', Usage, 'Render an unstyled list by default.')
  },
  variations() {
    return [
      snapshot(
        'Variants',
        Variants,
        'Lists can be displayed as plain list items or with dashes.'
      ),
      snapshot(
        'Colors',
        Colors,
        'Lists can be customized to use any of the Dracula UI theme colors.'
      )
    ]
  }
})

function Usage() {
  return (
    <List>
      <li className="drac-text drac-text-white">Blade</li>
      <li className="drac-text drac-text-white">Buffy</li>
      <li className="drac-text drac-text-white">Morbius</li>
    </List>
  )
}

function Variants() {
  return (
    <Box>
      <List color="purple">
        <li className="drac-text drac-text-white">Blade</li>
        <li className="drac-text drac-text-white">Buffy</li>
        <li className="drac-text drac-text-white">Morbius</li>
      </List>

      <List variant="unordered" color="purple">
        <li className="drac-text drac-text-white">Blade</li>
        <li className="drac-text drac-text-white">Buffy</li>
        <li className="drac-text drac-text-white">Morbius</li>
      </List>
    </Box>
  )
}

function Colors() {
  return (
    <List color="cyan">
      <li className="drac-text drac-text-white">Blade</li>
      <li className="drac-text drac-text-white">Buffy</li>
      <li className="drac-text drac-text-white">Morbius</li>
    </List>
  )
}
