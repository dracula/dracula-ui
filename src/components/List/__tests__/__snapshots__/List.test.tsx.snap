// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: List Basic Usage 1`] = `
Object {
  "docs": "Render an unstyled list by default.",
  "html": "<ol class=\\"drac-list\\">
  <li class=\\"drac-text drac-text-white\\">Blade</li>
  <li class=\\"drac-text drac-text-white\\">Buffy</li>
  <li class=\\"drac-text drac-text-white\\">Morbius</li>
</ol>
",
  "react": "<List>
  <li className=\\"drac-text drac-text-white\\">Blade</li>
  <li className=\\"drac-text drac-text-white\\">Buffy</li>
  <li className=\\"drac-text drac-text-white\\">Morbius</li>
</List>;
",
  "title": "Usage",
}
`;

exports[`Site: List Colors 1`] = `
Object {
  "docs": "Lists can be customized to use any of the Dracula UI theme colors.",
  "html": "<ol class=\\"drac-list drac-list-ordered drac-list-cyan\\">
  <li class=\\"drac-text drac-text-white\\">Blade</li>
  <li class=\\"drac-text drac-text-white\\">Buffy</li>
  <li class=\\"drac-text drac-text-white\\">Morbius</li>
</ol>
",
  "react": "<List variant=\\"ordered\\" color=\\"cyan\\">
  <li className=\\"drac-text drac-text-white\\">Blade</li>
  <li className=\\"drac-text drac-text-white\\">Buffy</li>
  <li className=\\"drac-text drac-text-white\\">Morbius</li>
</List>;
",
  "title": "Colors",
}
`;

exports[`Site: List Variants 1`] = `
Object {
  "docs": "Lists can be displayed in dashes as an \`<ul>\` or in numbers as an \`<ol>",
  "html": "<div class=\\"drac-box\\">
  <ol class=\\"drac-list drac-list-ordered drac-list-purple\\">
    <li class=\\"drac-text drac-text-white\\">Blade</li>
    <li class=\\"drac-text drac-text-white\\">Buffy</li>
    <li class=\\"drac-text drac-text-white\\">Morbius</li>
  </ol>
  <ul class=\\"drac-list drac-list-unordered drac-list-purple\\">
    <li class=\\"drac-text drac-text-white\\">Blade</li>
    <li class=\\"drac-text drac-text-white\\">Buffy</li>
    <li class=\\"drac-text drac-text-white\\">Morbius</li>
  </ul>
</div>
",
  "react": "<Box>
  <List variant=\\"ordered\\" color=\\"purple\\">
    <li className=\\"drac-text drac-text-white\\">Blade</li>
    <li className=\\"drac-text drac-text-white\\">Buffy</li>
    <li className=\\"drac-text drac-text-white\\">Morbius</li>
  </List>
  <List variant=\\"unordered\\" color=\\"purple\\">
    <li className=\\"drac-text drac-text-white\\">Blade</li>
    <li className=\\"drac-text drac-text-white\\">Buffy</li>
    <li className=\\"drac-text drac-text-white\\">Morbius</li>
  </List>
</Box>;
",
  "title": "Variants",
}
`;
